docs/bounded_dynamism.md

```diff
@@ -0,0 +1,85 @@
+# StableHLO Bounded Dynamism
```

Comment:
Similarly, can you rename the file to `rfcs/20220925-bounded-dynamism.md` similarly to what's proposed in #196? The convention is not final yet, but I think it would be good to already start separating docs and RFCs.